src/users/users.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@nestjs/common'
+import { Pageable } from 'src/interfaces/pageable.interface'
+import { QueryUserDto } from './dto/get-user.dto'
+import { UserEntity } from './entities/user.entity'
+import { UsersRepository } from './users.repository'
+
+@Injectable()
+export class UsersService {
+    constructor(private usersRepository: UsersRepository) {}
+
+    async getUserById(id: number, returnStudent: boolean): Promise<UserEntity> {
+        return await this.usersRepository.getUserById(id, returnStudent)
+    }
+
+    async getUsers(queryUserDto: QueryUserDto): Promise<Pageable<UserEntity>> {
+        const { page, pageSize, returnStudent, orderBy, orderField, ...filter } = queryUserDto
+
+        const except_fields = ['phone', 'nickname']
```

Comment:
move to constant in user module, change to `exceptFields` and declare type 
const exceptFields: (keyof QueryUserDto)[] = .....